src/on.ts

```diff
@@ -0,0 +1,93 @@
+import { Handle, EventObject } from './interfaces';
+import { createHandle, createCompositeHandle } from './lang';
+import Evented from './Evented';
+
+// Only used for Evented and EventEmitter, as EventTarget uses EventListener
+type EventCallback = (event: {}) => void;
+
+interface ExtensionEvent {
+	(target: any, listener: EventCallback): Handle;
+}
+
+interface EventTarget {
+	addEventListener(event: string, listener: EventListener, capture?: boolean): Handle;
+	removeEventListener(event: string, listener: EventListener, capture?: boolean): void;
+}
+
+interface EventEmitter {
+    on(event: string, listener: EventCallback): EventEmitter;
+	removeListener(event: string, listener: EventCallback): EventEmitter;
+}
+
+export default function on(target: EventTarget, type: string, listener: EventListener, capture?: boolean): Handle;
+export default function on(target: EventTarget, type: ExtensionEvent, listener: EventListener, capture?: boolean): Handle;
+export default function on(target: EventTarget, type: [string|ExtensionEvent], listener: EventListener, capture?: boolean): Handle;
```

Comment:
I'm OK with that, however that type statement is not doing what you think it is. `[string|ExtensionEvent]` is an array of one `string` or `ExtenstionEvent`. You probably want `(string | ExtensionEvent)[]`.